### PR TITLE
Add @since-annotations

### PR DESCRIPTION
**Pull request by sjakobi**
These annotations are based on the changelog and go back to v0.5.4 / GHC-7.8.

### CODE REVIEW

**Comment by sjakobi** [comment]
I just noticed that Travis isn't set up to build the haddocks – should that be changed?

**Comment by treeowl** [comment]
Wow! That is truly great work. Thank you so much! Do you think you could rebase it, and add the annotation I *should have* added to the `MonadFix Seq` instance?

**Comment by sjakobi** [comment]
> Do you think you could rebase it, and add the annotation I should have added to the MonadFix Seq instance?

Done.

I have also noticed that haddock reports a parse failure with GHC-8.0 for

```
  deriving ( Eq
           , Read
           , Show
           , Data
           , Generic  -- ^ @since 0.5.8
           , Generic1 -- ^ @since 0.5.8
           )
```

Not sure how to handle this best – conditional annotations with CPP or simply leave them off?

File: Data/IntMap/Internal.hs [hunk 1]
```diff
@@ -1815,7 +1872,7 @@ traverseMaybeWithKey f = go
 -- prop> symmetricDifference = merge diffPreserve diffPreserve (\ _ _ _ -> Nothing)
 -- prop> mapEachPiece f g h = merge (diffMapWithKey f) (diffMapWithKey g)
 --
--- @since 0.5.8
+-- @since 0.5.9
```
Comment (sjakobi) [inline Data/IntMap/Internal.hs]: I hope I didn't overlook anything here.

**Comment by sjakobi** [comment]
In addition to the rebase I have added annotations for the contents of the `Data.IntMap.Merge.*` modules.

What would be the right annotations for the contents of the `Data.Map.Merge.*` modules though? 0.5.9, when the modules were renamed, or 0.5.8, when they were first added?

**Comment by treeowl** [comment]
Ugh. I guess make them conditional. Do you know what went wrong?

**Comment by treeowl** [comment]
Urk. I don't know about the module renaming question. That might be one for
the libraries list. One makes it seem like functionality wasn't available
when it was; the other makes it seem like the module was there when it
wasn't.

**Comment by sjakobi** [comment]
> Ugh. I guess make them conditional. Do you know what went wrong?

Haddocks on derived instances (in the `deriving (…)` style) are a pretty new feature which required changes in `haddock` too. I guess old haddock parsers just can't handle the syntax.

**Comment by sjakobi** [comment]
> > What would be the right annotations for the contents of the Data.Map.Merge.* modules though? 
> > 0.5.9, when the modules were renamed, or 0.5.8, when they were first added?

> Urk. I don't know about the module renaming question. That might be one for
the libraries list. One makes it seem like functionality wasn't available
when it was; the other makes it seem like the module was there when it
wasn't.

@hvr Could you help us out with this question?

**Comment by treeowl** [comment]
You know, in this case, I think we should be conservative and use the later
version number. Almost no one used the earlier one, as it came between GHC
releases.

File: Data/IntMap/Internal.hs (hunk 1 above)
Comment (treeowl) [reply Data/IntMap/Internal.hs]: I probably just goofed up.

**Comment by treeowl** [comment]
Can I squash these, or do you want to preserve them as separate commits for some reason?

**Comment by sjakobi** [comment]
Squashing seems fine.

**Comment by hvr** [comment]
> Haddocks on derived instances (in the deriving (…) style) are a pretty new feature

iirc you can workaround this by using standalone deriving for backward compat; but I haven't tried

**Comment by treeowl** [comment]
@sjakobi, we can consider adding Haddocks to Travis, but I don't want to pay too much in CI time. Honestly, I don't think it's all that important for the docs to build successfully between releases. In any case, I'm merging this PR now, so that discussion should happen elsewhere. Thanks a lot for your work here.

**Pull request merged**